ionic/components/button/button.md.scss

```diff
@@ -58,7 +58,7 @@ $button-md-small-icon-font-size:           1.4em !default;
               color $button-md-transition-duration $button-md-animation-curve;
 
   &:hover:not(.disable-hover) {
-    background-color: $button-md-clear-hover-background-color;
+    background-color: $button-md-color;
```

Comment:
yeah ::before can be used if you don't want the text to lighten, but in iOS mode the whole button goes to opacity: 0.9 which lightens the text on hover but doesn't seem weird or problematic so I would think either is fine 

I'm still convinced `:hover { $button-md-color }` here is redundant (i.e. the whole rule can be removed) since the base style without `:hover` is already that 

and in theory maybe this could cause issues (e.g. of a `<button secondary>` being hovered with this rule does the button suddenly turn the `<button primary>` color on hover?) but I guess we'll find out :smile: